content/gocourse-2019-05-melbourne.slide

```diff
@@ -1142,6 +1142,35 @@ Rob Pike's Talk ([[https://blog.golang.org/toward-go2][Towards Go 2]])
 Dave Cheney Blog ([[https://dave.cheney.net/paste/the-past-present-and-future-of-go2.pdf][Past Present and Future of Go]])
 
 
+* Go for Java devs
+
+- Functions are lambdas
+- Typed constants are enums
+- exported = public
+- unexported = protected, not private
+- struct != class
```

Comment:
Can you elaborate in a couple of sentences? do you mean because structs can exist without methods? because there is no inheritance and no implements? because there is no such thing as class vs instance method? maybe add speaker notes?

Reply:
I was thinking about how to elaborate on this more, there are numerous nuanced differences, things like there are no static variables/methods (this means the equivalent of a utility class in java is actually a utility package in go)
I suppose the point I'm trying to make is that java devs won't use struct as widely or as varied as they will use class.
Things that are difference
* No concept of static/final
* No constructors
* No inheritence (embedding is not extending)
* No implements
* No class level variable/method scope (scope is at most package level)
* Functions do not need to be tied to a class

I figured that was too much detail and starting to go into too much detail, I'll add speaker notes for this point. Not sure about examples because these points are mainly what you don't use structs for. Thoughts?

Reply:
Fair enough, might be worthwhile to add as speaker notes though.
Alternatively you could add separate slide for `struct != class`